### PR TITLE
[recipe, ckpt] fix: align training config for NPU and fix OOM during …

### PR DESCRIPTION
**Pull request by panshaowu**
…checkpointing

### What does this PR do?

> Add **concise** overview of what this PR aims to achieve or accomplish. Reference related GitHub issues and PRs that help with the review.
This PR improves the Ascend NPU training experience for Qwen-Image FlowGRPO by:
   1. Parameter Alignment: Updated the NPU training hyperparameters (batch size, learning rate, and PPO micro-batches) in run_qwen_image_ocr_lora_npu.sh to match the standard GPU training recipe, ensuring consistency in training behavior and throughput.
   2. OOM Mitigation: Revised the memory release mechanism in DiffusersFSDPEngine. By introducing aggressive_empty_cache during the checkpoint saving process, it ensures fragmented memory is reclaimed on NPU, preventing OOM failures during heavy checkpointing operations.

### Checklist Before Starting

- [x] Search for similar PRs. Paste at least one query link here: ...
- [x] Format the PR title as `[{modules}] {type}: {description}` (This will be checked by the CI)
  - `{modules}` include `fsdp`, `vllm_omni`, `rollout`, `trainer`, `ci`, `training_utils`, `recipe`, `ray`, `worker`, `single_controller`, `misc`, `perf`, `model`, `algo`, `env`, `tool`, `ckpt`, `doc`, `data`, `cfg`, `reward`, `diffusion`, `omni`, `tests`, `docker`
  - If this PR involves multiple modules, separate them with `,` like `[diffusion, doc]`
  - `{type}` is in `feat`, `fix`, `refactor`, `chore`, `test`
  - If this PR breaks any API (CLI arguments, config, function signature, etc.), add `[BREAKING]` to the beginning of the title.
  - Example: `[BREAKING][diffusion, fsdp] feat: new rollout scheduler`

### Test

> For changes that can not be tested by CI (e.g., algorithm implementation, new model support), validate by experiment(s) and show results like training curve plots, evaluation results, etc.

Validated on Atlas 800T A2 (CANN 9.0.0):
   1. Smoke Tests: All tests in tests/npu_smoke/run_npu_smoke_tests.sh passed.
   2. E2E Training: Successfully ran OCR LoRA training using examples/flowgrpo_trainer/run_qwen_image_ocr_lora_npu.sh with 8 NPUs (TP8 strategy), confirming stable memory usage and convergence logic.
   3. The reward value at step `0`, `30`, and `60` are `0.9052`, `0.9252`, and `0.9691`. The loss and reward curves are as follows:

image

### API and Usage Example

> Demonstrate how the API changes if any, and provide usage example(s) if possible.

```shell
bash examples/flowgrpo_trainer/run_qwen_image_ocr_lora_npu.sh
```

### Design & Code Changes

> Demonstrate the high-level design if this PR is complex, and list the specific changes.

- Recipe Update: Synced examples/flowgrpo_trainer/run_qwen_image_ocr_lora_npu.sh with run_qwen_image_ocr_lora.sh to provide a unified baseline for performance comparison.
- Memory Optimization: In verl_omni/workers/engine/fsdp/diffusers_impl.py, modified save_checkpoint to trigger aggressive_empty_cache(force_sync=True) after FSDP parameters are offloaded to CPU. This proactively clears the NPU allocator's cache to handle the peak memory demand during state dict processing.

### Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [x] Read the [Contribute Guide](https://github.com/verl-project/verl-omni/blob/main/CONTRIBUTING.md).
- [x] Apply [pre-commit checks](https://github.com/verl-project/verl-omni/blob/main/CONTRIBUTING.md#code-linting-and-formatting): `pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always`
- [x] Add / Update the documentation.
- [x] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/verl-project/verl-omni/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: ...

### CODE REVIEW

**Review by zhtmike — APPROVED** [review]
LGTM

**Review by SamitHuang — APPROVED** [review]
LGTM
